src/aat/java/uk/gov/hmcts/ccd/datastore/tests/functional/UpdateCaseTest.java

```diff
@@ -68,4 +68,188 @@ void shouldUpdateSingleField() {
+    }
+
+    @Test
+    @DisplayName("should not update a case if the caseworker has 'CR' access on CaseType")
+    void shouldNotUpdateCaseWithCRAccessForCaseType() {
+        // Case Type with "CR" access to the role autoTestCaseWorker
+        final Long caseReference = Event.create("AAT_AUTH_3")
+            .as(asPrivateCaseworker(true))
+            .withData(FullCase.build())
+            .submitAndGetReference();
+
+        Event.update(caseReference)
+            .as(asAutoTestCaseworker())
+            .withData(
+                CaseData.builder()
+                    .numberField(UPDATED_NUMBER)
+                    .build()
+            )
+            .submit()
+
+            .then()
+            .statusCode(404);
+    }
+
+    @Test
+    @DisplayName("should not update a case if the caseworker has 'C' access on CaseType")
+    void shouldNotUpdateCaseWithCAccessForCaseType() {
+        // Case Type with "C" access to the role autoTestCaseWorker
+        final Long caseReference = Event.create("AAT_AUTH_1")
+            .as(asPrivateCaseworker(true))
+            .withData(FullCase.build())
+            .submitAndGetReference();
+
+        Event.update(caseReference)
+            .as(asAutoTestCaseworker())
+            .withData(
+                CaseData.builder()
+                    .numberField(UPDATED_NUMBER)
+                    .build()
+            )
+            .submit()
+
+            .then()
+            .statusCode(404);
+    }
+
+    @Test
+    @DisplayName("should not update a case if the caseworker has 'R' access on CaseType")
+    void shouldNotUpdateCaseWithRAccessForCaseType() {
+        // Case Type with "R" access to the role autoTestCaseWorker
+        final Long caseReference = Event.create("AAT_AUTH_2")
+            .as(asPrivateCaseworker(true))
+            .withData(FullCase.build())
+            .submitAndGetReference();
+
+        Event.update(caseReference)
+            .as(asAutoTestCaseworker())
+            .withData(
+                CaseData.builder()
+                    .numberField(UPDATED_NUMBER)
+                    .build()
+            )
+            .submit()
+
+            .then()
+            .statusCode(404);
+    }
+
+    @Test
+    @DisplayName("should not update a case if the caseworker has 'D' access on CaseType")
+    void shouldNotUpdateCaseWithDAccessForCaseType() {
+        // Case Type with "D" access to the role autoTestCaseWorker
+        final Long caseReference = Event.create("AAT_AUTH_8")
+            .as(asPrivateCaseworker(true))
+            .withData(FullCase.build())
+            .submitAndGetReference();
+
+        Event.update(caseReference)
+            .as(asAutoTestCaseworker())
+            .withData(
+                CaseData.builder()
+                    .numberField(UPDATED_NUMBER)
+                    .build()
+            )
+            .submit()
+
+            .then()
+            .statusCode(404);
+    }
```

Comment:
We should verify that another user with enough access to this case cen see this case, as a verification of impossible deletion despite existence of the case.

Reply:
Mutlu,

  In the same test method, I had infact created and retrieved the case data using the super user (private case worker). So, this already proves that another user have access to read the case data. Please confirm if that fine or I need to add a code snippet to update the same case super user. I thought that is not needed as we already have other test methods to test those scenarios.